spring-ai-core/src/main/java/org/springframework/ai/model/tool/ToolCallingChatOptions.java

```diff
@@ -53,12 +53,12 @@ public interface ToolCallingChatOptions extends FunctionCallingOptions {
 	/**
 	 * Names of the tools to register with the ChatModel.
 	 */
-	Set<String> getTools();
+	Set<String> getToolNames();
```

Comment:
Having the same API (`tools()`) in `ChatClient` and `ToolCallingChatOptions`, but with very different behaviours, didn't seem good.

Now, the `ChatClient` keeps having one single API with overloaded methods to support different use cases: `tools()`.
`ToolCallingChatOptions`, instead, reflects the two types of tool-related information it stores: `toolNames()` and `toolCallbacks()`.

@tzolov what do you think?